src/arch/amdgpu/vega/insts/op_encodings.hh

```diff
@@ -421,6 +422,145 @@ namespace VegaISA
         InstFormat extData;
         uint32_t varSize;
 
+        template<typename T>
+        uint32_t
+        sdwabSelect(uint32_t dword, const SDWASelVals sel,
+                    bool sign_ext, bool neg, bool abs)
+        {
+            int low_bit = 0, high_bit = 0;
+            uint32_t rv = dword;
+
+            if (sel < SDWA_WORD_0) {
```

Comment:
Ideally we should add comments explaining what each of these are doing.

Reply:
Done